File: leetcode/26-RemoveDuplicatesFromSortedArray/26-RemoveDuplicatesFromSortedArray.py
# Last updated: 29/04/2026, 23:14:03
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        l = 0
        r = 0
        while l < len(nums) and r < len(nums):
            if nums[l] != nums[r]:
                nums[l+1] = nums[r]
                l += 1
            r += 1
        return l + 1

File: leetcode/26-RemoveDuplicatesFromSortedArray/26-RemoveDuplicatesFromSortedArray.cpp
// Last updated: 29/04/2026, 23:14:08
class Solution
{
public:
    int removeDuplicates(vector<int> &nums)
    {
        if (nums.size() == 0)
            return 0;
        int repeated = 0;
        for (int i = 0; i < nums.size() - 1; i++)
            if (nums[i] == nums[i + 1])
            {
                repeated++;
                nums.erase(nums.begin() + i + 1);
                i--;
            }
        return nums.size();
    }
};